src/main/java/httpserver/HttpResponse.java

```diff
@@ -0,0 +1,365 @@
+                noContent();
+            }
+
+            // Send the required headers down the pipe.
+            writeLine("HTTP/1.1 " + getResponseCodeMessage(getCode()));
+            writeLine("Server: " + getServerInfo());
+            writeLine("Content-Type: " + getMimeType());
+
+            writeLine("Connection: close");
+
+            if (getSize() != -1) {
+                // Someone manually set the size of the body. Go team!
+                writeLine("Content-Size: " + getSize());
+            } else {
+                // We don't know how large the body is. Determine that using the body...
+                writeLine("Content-Size: " + getBody().length);
+            }
+
+            // Send all other miscellaneous headers down the shoots.
+            if (!getHeaders().isEmpty()) {
+                StringBuilder b = new StringBuilder();
+                for (String key : getHeaders().keySet()) {
+                    b.append(key);
+                    b.append(": ");
+                    b.append(getHeader(key));
+                    b.append("\n");
+                }
+                writeLine(b.toString());
+            }
+
+            // Blank line separating headers from the body.
+            writeLine("");
+
+            // If there isn't a body, or the client made a HEAD request, stop
+            // doing things.
+            if (getRequest().isType(HttpRequest.HEAD_REQUEST_TYPE) || getCode() == 204) {
+                return;
+            }
+
+            // Give the client the body.
+            getWriter().write(getBody());
+        } catch (HttpException | IOException e) {
+            System.err.println("Something bad happened while trying to send data "
+                    + "to the client");
+            e.printStackTrace();
+        } finally {
+            try {
+                getWriter().close();
+            } catch (NullPointerException | IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Writes a string and a "\n" to the DataOutputStream.
+     * @param line The line to write
+     * @throws IOException
+     */
+    protected void writeLine(String line) throws IOException {
+        getWriter().writeBytes(line + "\n");
+    }
+
+
+    /*********************
+      GETTERS AND SETTERS
+     *********************/
+
+    public int getCode() {
+        return code;
+    }
+    public void setCode(int code) {
+        this.code = code;
+    }
+
+
+    public byte[] getBody() {
+        return body;
+    }
+    public void setBody(String body) {
+        this.body = body.getBytes();
+    }
+    public void setBody(byte[] bytes) {
+        body = bytes;
+    }
+
+
+    public String getMimeType() {
+        return mimeType;
+    }
+    public void setMimeType(String mimeType) {
+        this.mimeType = mimeType;
+    }
+
+
+    public long getSize() {
+        return size;
+    }
+    public void setSize(long size) {
```

Comment:
This should probably have some sort of check to make sure you don't say the content size is something like -400, because it can't be less than 0.

Alternatively, change line 149 to `if (getSize() >= 0)`.